packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -206,10 +207,17 @@ function FlowsheetDataGrid(props) {
     return newSections;
   }, [intl, sections]);
 
+  const flattenSections = (sectionsToFlatten) => sectionsToFlatten.flatMap(section => section.rows.map(row => ({
+    ...row,
+    sectionId: section.id,
+  })));
```

Comment:
If you needed to flatten the sections, wouldn't you want to memoize the value?  It looks like you are calling this method in multiple places.  However, the same value should be returned.